pkg/k8s_utils/k8s_client.go

```diff
@@ -30,12 +39,64 @@ type KubernetesClient struct {
 }
 
 // GetNameSpace ...
-func (kc *KubernetesClient) GetNameSpace() string {
+func (kc KubernetesClient) GetNameSpace() string {
 	kc.logger.Info("Fetching namespace")
 	return kc.namespace
 }
 
 // GetClientSet ...
-func (kc *KubernetesClient) GetClientSet() kubernetes.Interface {
+func (kc KubernetesClient) GetClientSet() kubernetes.Interface {
 	return kc.clientset
 }
+
+// Getk8sClientSet ...
+func Getk8sClientSet(logger *zap.Logger) (KubernetesClient, error) {
+	logger.Info("Fetching k8s clientset")
+
+	var kc KubernetesClient
+	// Fetching cluster config used to create k8s client
+	k8sConfig, err := rest.InClusterConfig()
+	if err != nil {
+		logger.Error("Error fetching in cluster config", zap.Error(err))
+		return kc, utils.Error{Description: utils.ErrFetchingK8sClusterConfig, BackendError: err.Error()}
+	}
+
+	// Creating k8s client used to read secret
+	clientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		logger.Error("Error creating k8s client", zap.Error(err))
+		return kc, utils.Error{Description: utils.ErrFetchingK8sClusterConfig, BackendError: err.Error()}
+	}
+	logger.Info("Successfully fetched k8s client set")
+
+	namespace, err := getNameSpace(logger)
+	if err != nil {
+		logger.Error("Error fetching namespace", zap.Error(err))
+		return kc, err
+	}
+	logger.Info("Successfully fetched namespace")
+
+	kc.clientset = clientset
+	kc.logger = logger
+	kc.namespace = namespace
+	return kc, nil
+}
+
+// getNameSpace ...
+func getNameSpace(logger *zap.Logger) (string, error) {
+	// Reading the namespace in which the pod is deployed
```

Comment:
`Reading the namespace in which the pod is deployed`
which pod?

Reply:
CSI driver